rclpy/src/rclpy/timer.hpp

```diff
@@ -17,127 +17,141 @@
 
 #include <pybind11/pybind11.h>
 
+#include <rcl/timer.h>
+
+#include <memory>
+
+#include "destroyable.hpp"
+#include "handle.hpp"
+
 namespace py = pybind11;
 
 namespace rclpy
 {
-/// Reset the timer
-/**
- * Raise ValueError if capsule is not a timer
- * Raises RCLError if the timer cannot be reset
- *
- * \param[in] pytimer Capsule pointing to the timer
- */
-void
-reset_timer(py::capsule pytimer);
 
-/// Checks if timer reached its timeout
-/**
- *  Raises ValueError if pytimer is not a timer capsule
- *  Raises RCLError if there is an rcl error
- *
- * \param[in] pytimer Capsule pointing to the timer
- * \return True if the timer is ready
- */
-bool
-is_timer_ready(py::capsule pytimer);
+class Timer : public Destroyable
+{
+public:
+  /// Create a timer
+  /**
+   * This class will create a timer
+   *
+   * Raises RCLError on initialization failure
+   * Raises TypeError if argument of invalid type
+   * Raises ValueError if argument cannot be converted
+   *
+   * \param[in] clock pycapsule containing an rcl_clock_t
+   * \param[in] pycontext Capsule for an rcl_timer_t
+   * \param[in] period_nsec the period of the timer in nanoseconds
+   * \return a timer capsule
+   */
+  Timer(py::capsule pyclock, py::capsule pycontext, int64_t period_nsec);
 
-/// Set the last call time and start counting again
-/**
- * Raises ValueError if pytimer is not a timer capsule
- * Raises RCLError if there is an rcl error
- *
- * \param[in] pytimer Capsule pointing to the timer
- */
-void
-call_timer(py::capsule pytimer);
+  ~Timer() = default;
 
-/// Update the timer period
-/**
- * The change in period will take effect after the next timer call
- *
- * Raises ValueError if pytimer is not a timer capsule
- * Raises RCLError if the timer period could not be changed
- *
- * \param[in] pytimer Capsule pointing to the timer
- * \param[in] period_nsec the new period in nanoseconds
- */
-void
-change_timer_period(py::capsule pytimer, int64_t period_nsec);
+  /// Reset the timer
+  /**
+   * Raise ValueError if capsule is not a timer
```

Comment:
I think you can delete this line since `timer` is no longer an argument

Reply:
Same comment about deleting the `ValueError` comment on most of the functions below